### ISSUE TITLE
dc-tool-ser does not work if compiled with BAUD_RATE != 57600

### ISSUE THREAD

**Issue by darcagn** [KallistiOS/dcload-serial#10]
Whether you set `SERIALSPEED =` in Makefile.cfg or `-DBAUD_RATE=` in the CFLAGs in `host-src/tool/Makefile`, dc-tool-ser does not work (at all, no matter what your runtime arguments are) unless compiled with the default value `57600`. 

For example, if I compile with it with `38400`, `115200`, `1500000`, etc. it will display the following, and then just hang:
```
Console enabled
Upload </home/darc/dc-code/dc-mboot/bootcd.elf>
File format is ELF, start address is 0x8ce00000
Section .text, lma 0x8ce00000, size 29640
```

It doesn't seem to have anything to do with the actual FTDI USB adapter or Dreamcast.
As if I compile with `115200` or `1500000` and then set the baud rate back to `57600` using `-b 57600`, it doesn't work.
But if I compile with `57600` and set the baud rate to `1500000` using `-b 1500000`, it works at 1.5M speed.

**Comment by pcercuei** [comment]
IIRC you should keep SERIALSPEED to the default and only update BAUD_RATE in your CFLAGS (or the other way around).

**Comment by darcagn** [comment]
`BAUD_RATE` in the CFLAGS is set to whatever `SERIALSPEED` is in `Makefile.cfg`. Originally I set it in the `Makefile.cfg` (as is the proper way to do it), but later set it directly in the CFLAGS as it was easier to modify there and just compile the host tool Makefile alone as I was recompiling over and over trying to find the source of the problem. No matter which way you do it though the bug manifests itself.

**Comment by pcercuei** [comment]
```diff
diff --git a/Makefile.cfg b/Makefile.cfg
index 349b217..171c6eb 100644
--- a/Makefile.cfg
+++ b/Makefile.cfg
@@ -47,7 +47,7 @@ TARGETOBJCOPY = $(TARGETPREFIX)/bin/sh-elf-objcopy -R .stack
 TARGETLD       = $(TARGETPREFIX)/bin/sh-elf-ld
 
 # this is the default port for dc-tool, it can be changed on the commandline
-SERIALDEVICE   = /dev/ttyS0
+SERIALDEVICE   = /dev/ttyUSB0
 
 # mac osx -- you'll need to go look in /dev to see what the actual device
 # name is, and depending on the device, it must be plugged into the same
@@ -64,6 +64,6 @@ TOOLINSTALLDIR        = /opt/toolchains/dc/bin
 #EXECUTABLEEXTENSION = .exe
 
 
-# default serial speed - YOU DO NOT NEED TO CHANGE THIS
-SERIALSPEED    = 57600
+# default serial speed
+SERIALSPEED    = 1500000
 
diff --git a/host-src/tool/dc-tool.c b/host-src/tool/dc-tool.c
index aff998a..d0f2124 100644
--- a/host-src/tool/dc-tool.c
+++ b/host-src/tool/dc-tool.c
@@ -1232,10 +1232,10 @@ int main(int argc, char *argv[])
         close_serial();
     }
 
-    if (open_serial(device_name, BAUD_RATE, &dummy)<0)
+    if (open_serial(device_name, 57600, &dummy)<0)
         return 1;
 
-    if (speed != BAUD_RATE)
+    if (speed != 57600)
        change_speed(device_name, speed);
 
     switch (command) {

```

**Comment by pcercuei** [comment]
This above is my local diff (from ages ago). I believe a correct patch would set `BAUD_RATE` in the Makefile to 57600.

**Comment by darcagn** [comment]
If I apply that patch, it simply hangs at `Console enabled`. If I then set `BAUD_RATE` in the Makefile to 57600, then it transfers at 57600 baud.

**Comment by pcercuei** [comment]
If you set `BAUD_RATE` to 57600 and `SERIALSPEED` to `1500000` then it transfers at 57600 baud?

I'm at f2ce242 if that matters, and it works here.

**Comment by darcagn** [comment]
Correct, and unless I'm misunderstanding you, I don't see how any other behavior can be expected given that `SERIALSPEED` is never used it in the host tool, except for the Makefile to set `BAUD_RATE` to its value. If I remove that and hardcode `BAUD_RATE` to `57600`, then the desired `1500000` value is never being passed to the code at all?

**Comment by pcercuei** [comment]
Yeah, sorry. Got confused.

What works for me is `SERIALSPEED` to `1500000` with the patch above. Doesn't hang here.

**Comment by darcagn** [comment]
I think we were both confused, because I manually applied the patch with a text editor and accidentally modified the wrong part. 

I started over from a fresh clone and if I simply make the two `BAUD_RATE`->`57600` changes in `dc-tool.c` (and don't touch the `Makefile` at all) then the `SERIALSPEED` setting in `Makefile.cfg` properly alters the behavior to the desired behavior. 👍

**Comment by ljsebald** [comment]
Closed for the reasoning explained in PR #11.
